set_bucket_tags/app.py

```diff
@@ -54,24 +53,34 @@ def get_principal_id(tags):
     raise ValueError('Could not derive a provisioningPrincipalArn from tags')
 
 
-def get_synapse_email(synapse_id):
-  '''Derive an email from a id'''
+def get_synapse_user_profile(synapse_id):
+  '''Get synapse user profile data'''
   syn = synapseclient.Synapse()
   user_profile = syn.getUserProfile(synapse_id)
   log.debug(f'Synapse user profile: {user_profile}')
-  user_name = user_profile.get('userName')
-  synapse_email = f'{user_name}@synapse.org'
-  return synapse_email
 
+  return user_profile
 
-def add_owner_email_tag(tags, email):
-  '''Add an OwnerEmail tag to set of bucket tags'''
-  owner_email_tag = next((tag for tag in tags if tag['Key'] == 'OwnerEmail'), None)
-  if owner_email_tag:
-    owner_email_tag['Value'] = email
-  else:
-    new_owner_email_tag = { 'Key': 'OwnerEmail', 'Value': email}
-    tags.append(new_owner_email_tag)
+
+def get_synapse_tags(user_profile):
+  '''Derive synapse tags from synapse user profile data'''
+
+  tags = []
+  IGNORE_KEYS = ["createdOn"]
+  for key, value in user_profile.items():
+    if key in IGNORE_KEYS:
+      continue
+
+    # derive synapse email tag based on userName
+    if key == "userName":
+      synapse_email = f'{value}@synapse.org'
+      tags.append({'Key': 'synapse:email', 'Value': synapse_email})
+      tags.append({'Key': 'OwnerEmail', 'Value': synapse_email})  # legacy
+
+    tag = {'Key': f'synapse:{key}', 'Value': value}
```

Comment:
Remember that any tag that that doesn't start with `Protected/` can be overwritten by any other instance.  IMO these should all be under the `Protected/` tag path *or* we remove the policy for instances to apply tags to themselves (is there another PR that does that?)